src/openms/source/FORMAT/FASTAFile.cpp

```diff
@@ -39,179 +39,240 @@
-      {
-        break;
-      }
-      firstline = infile_.tellg();
-    }
-    infile_.seekg(firstline);
-  
-    // automatically deletes old handles
-    reader_ = std::unique_ptr<void, std::function<void(void*) > >(new FASTARecordReader(infile_),
-      [](void* ptr)
-      { // lambda with custom cast
-      delete static_cast<FASTARecordReader*>(ptr);
-      });
-
-    entries_read_ = 0;
-  }
-
-  bool FASTAFile::readNext(FASTAEntry& protein)
-  {
-    if (seqan::atEnd(*static_cast<FASTARecordReader*>(reader_.get())))
-    { 
-      // do NOT close(), since we still might want to seek to certain positions
-      return false;
+       std::string line;
+       if(TextFile::getLine(infile_,line))//using Textfile::getLine to be able to read '>'
+       {
+           line.erase(0,1);
+           id=line;
+       }
+       else return false; //infile_ empty
+
+        while(FASTAFile::getLine_(infile_, line))
+        {
+            seq+=line;
+        }
+        seq+=line;//after getLine_ returns false still add the line read
+
+        if(seq.empty()) return false;
+        return true;
     }
-    String id, s;
-    if (readRecord(id, s, *static_cast<FASTARecordReader*>(reader_.get()), seqan::Fasta()) != 0)
+
+    void FASTAFile::readStart(const String& filename)
     {
-      if (entries_read_ == 0) s = "The first entry could not be read!";
-      else s = "Only " + String(entries_read_) + " proteins could be read. The record after failed.";
-      throw Exception::ParseError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "", "Error while parsing FASTA file! " + s + " Please check the file!");
+        if (!File::exists(filename))
+        {
+            throw Exception::FileNotFound(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, filename);
+        }
+
+        if (!File::readable(filename))
+        {
+            throw Exception::FileNotReadable(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, filename);
+        }
+
+        if (infile_.is_open()) infile_.close(); // precaution
+
+        infile_.open(filename.c_str(), std::ios::binary | std::ios::in);
+
+        infile_.seekg(0, infile_.end);
+        fileSize_ = infile_.tellg();
+        infile_.seekg(0, infile_.beg);
+
+        // Skip the header of PEFF files (http://www.psidev.info/peff)
+        std::string line;
+        while (FASTAFile::getLine_(infile_, line))
+        {
+            //skipping PEFF header or anything before the first identifier
+        }
+
+        entries_read_ = 0;
     }
-    ++entries_read_;
-    s.removeWhitespaces();
-    protein.sequence = s; // assign here, since 's' might have higher capacity, thus wasting memory (usually 10-15%)
-
-    // handle id
-    id.trim();
-    String::size_type position = id.find_first_of(" \v\t");
-    if (position == String::npos)
+
+    bool FASTAFile::readNext(FASTAEntry& protein)
     {
-      protein.identifier = std::move(id);
-      protein.description = "";
+        if (infile_.eof())
```

Comment:
maybe just `if (infile_)`, to make sure that you also test of the filestream has other issues (not sure if `eof` covers those).